packages/ui/src/components/input/date/index.tsx

```diff
@@ -37,16 +44,36 @@ export const DateInput = forwardRef<HTMLInputElement, DateInputProps>(
         const [anchorEl, setAnchorEl] = useState<HTMLInputElement | null>(null);
         const popoverRef = useRef<HTMLDivElement>(null);
         const [open, setOpen] = useState(false);
+        const [modal, setModal] = useState(false);
+
+        const resizeObserver = useRef(
+            new ResizeObserver((entries) => {
+                const { width } = entries[0].contentRect;
+                if (width < 640) setModal(true);
+                else setModal(false);
+            })
+        );
 
         const resolvedId = id || generatedId;
 
         useClickAway(popoverRef, () => {
             setOpen(false);
         });
 
+        useEffect(() => {
+            resizeObserver.current.observe(document.body);
+            return () => {
+                // eslint-disable-next-line react-hooks/exhaustive-deps
+                resizeObserver.current.unobserve(document.body);
+            };
+        });
```

Comment:
this useEffect is the same as above, as it is the `resizeObserver = useRef`

Maybe there is an opportunity to extract it.

Reply:
I'd leave it like this for now, since it's just a single `useEffect`.